crates/typst-syntax/src/lexer.rs

```diff
@@ -687,6 +699,56 @@ impl Lexer<'_> {
     }
 }
 
+/// Math argument list.
+impl Lexer<'_> {
+    fn math_args(&mut self, start: usize, c: char) -> (SyntaxKind, Option<SyntaxNode>) {
+        match c {
+            c if self.is_math_named_arg(start, c) => (SyntaxKind::Ident, None),
+            '.' if self.is_math_spread_arg() => (SyntaxKind::Dots, None),
+            _ => self.math(start, c),
+        }
+    }
+
+    /// Handle named arguments in math function call.
+    fn is_math_named_arg(&mut self, start: usize, c: char) -> bool {
```

Comment:
You can write this a lot shorterif you swap the boolean conditions  on the if-statements (e.g. `!is_id_start() -> is_id_start()`) and just `return true` in one spot.